arcgis-ios-sdk-samples/Utility network/Create load report/CreateLoadReportViewController.swift

```diff
@@ -0,0 +1,360 @@
+// Copyright 2021 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ArcGIS
+
+class CreateLoadReportViewController: UIViewController {
+    // MARK: Storyboard views
+    
+    /// The table view to display the load reports.
+    @IBOutlet var tableView: UITableView!
+    @IBOutlet var resetBarButtonItem: UIBarButtonItem!
+    @IBOutlet var runBarButtonItem: UIBarButtonItem!
+    
+    // MARK: Properties
+    
+    /// A feature service for an electric utility network in Naperville, Illinois.
+    let utilityNetwork = AGSUtilityNetwork(url: URL(string: "https://sampleserver7.arcgisonline.com/arcgis/rest/services/UtilityNetwork/NapervilleElectric/FeatureServer")!)
+    /// The initial conditional expression.
+    var initialExpression: AGSUtilityTraceConditionalExpression!
+    /// The trace parameters for creating load reports.
+    var traceParameters: AGSUtilityTraceParameters!
+    /// The network attributes for the comparison.
+    var phasesNetworkAttribute: AGSUtilityNetworkAttribute!
+    /// A list of possible phases populated from the network's attributes.
+    var excludedPhases = [AGSCodedValue]()
+    /// A list of phases that are included in the load report.
+    var includedPhases = [AGSCodedValue]() {
+        didSet {
+            let isEmpty = includedPhases.isEmpty
+            DispatchQueue.main.async { [weak self] in
+                self?.runBarButtonItem.isEnabled = !isEmpty
+            }
```

Comment:
Having to dispatch async in a property observer like this is a bit of a code smell. When the dispatched work is executed, there's no guarantee that `isEmpty == includedPhases.isEmpty`. Almost always, it would be better to move the dispatch to the place where the property is being updated. In this case, dispatching wouldn't be needed if the summaries were accumulated into a local variable and then the property was updated in the `work` closure of the call to `notify(qos:flags:queue:execute:)`, which is called on the main thread.